### PR TITLE
Style: Disallow using DEPRECATED settings

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Style: Disallow using DEPRECATED settings.

Will need more interesting changes in private.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit f3e64fd69c0dfed3381a42c26fcc29819bd2fb4c with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71148/f3e64fd69c0dfed3381a42c26fcc29819bd2fb4c/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/71068/93494b08f594530e913ecf59507b24569704640f/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
The CI is stuck. Please update and also ask @maxknv

**Comment by Algunenano** [comment]
It's ok, the private sync PR is blocked by other pending PR to be able to pass the new style check and pass CI

File: src/Processors/QueryPlan/AggregatingStep.cpp [hunk 1]
```diff
@@ -32,8 +32,8 @@ namespace DB
 namespace QueryPlanSerializationSetting
 {
     extern const QueryPlanSerializationSettingsUInt64 aggregation_in_order_max_block_bytes;
-    extern const QueryPlanSerializationSettingsUInt64 aggregation_in_order_memory_bound_merging;
-    extern const QueryPlanSerializationSettingsUInt64 aggregation_sort_result_by_bucket_number;
+    extern const QueryPlanSerializationSettingsBool aggregation_in_order_memory_bound_merging;
```
Comment (Algunenano) [inline src/Processors/QueryPlan/AggregatingStep.cpp:35]: This was a bug introduced recently

**Comment by Algunenano** [comment]
Failures: https://github.com/ClickHouse/ClickHouse/issues/73639 https://github.com/ClickHouse/ClickHouse/issues/73638

**Pull request merged**